### PR TITLE
Fix syncSetup subpage was set after finishing search in settings

### PR DESCRIPTION
**Pull request by simonhong**
Removed all syncSetup templates

Resolves https://github.com/brave/brave-browser/issues/10506

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
See STR in issue

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: browser/resources/settings/brave_overrides/people_page.js [hunk 1]
```diff
@@ -32,6 +32,18 @@ RegisterPolymerTemplateModifications({
       console.error('[Brave Settings Overrides] People Page cannot find signin template')
       return
     }
+    const syncSetupLink = templateContent.querySelector('#sync-setup')
+    if (syncSetupLink) {
+      syncSetupLink.parentNode.removeChild(syncSetupLink)
```
Comment (petemill) [inline browser/resources/settings/brave_overrides/people_page.js:37]: Is there a reason why not `syncSetupLink.remove()` for all of these?

Comment (simonhong) [reply browser/resources/settings/brave_overrides/people_page.js]: Oh, I didn't know `remove()` :) All replaced with it. thanks!

**Comment by simonhong** [comment]
Merged because only `test-install` step was failed.

**Comment by kjozwiak** [comment]
Reproduced the original issue using the following build:

```
Brave | 1.12.85 Chromium: 84.0.4147.85 (Official Build) dev (64-bit)
-- | --
Revision | e20c13e958edc3707db3d26d513780a7f6331a4b-refs/branch-heads/4147@{#837}
OS | macOS Version 10.15.5 (Build 19F101)
```

Verification **PASSED** on `macOS 10.15.5 x64` using the following build:

```
Brave | 1.13.10 Chromium: 84.0.4147.85 (Official Build) nightly (64-bit)
-- | --
Revision | e20c13e958edc3707db3d26d513780a7f6331a4b-refs/branch-heads/4147@{#837}
OS | macOS Version 10.15.5 (Build 19F101)
```

* Verified using the STR outlined via https://github.com/brave/brave-browser/issues/10506.

**Pull request merged**